homeassistant/components/daikin/__init__.py

```diff
@@ -72,6 +72,8 @@ async def async_setup_entry(hass: HomeAssistantType, entry: ConfigEntry):
     # For backwards compat, set unique ID
     if entry.unique_id is None:
         hass.config_entries.async_update_entry(entry, unique_id=conf[KEY_MAC])
+    elif ".local" in entry.unique_id:
+        hass.config_entries.async_update_entry(entry, unique_id=conf[KEY_MAC])
```

Comment:
The existing config entries that were saved via zeroconf don't have the mac address in their entry data, right?

Reply:
Yes they do, the get the mac set in https://github.com/home-assistant/core/pull/36704/files/39b0bfa1141499afa725f15b4ded574871b18225#diff-9ba0b5cb03cb9763543902445049c666R52

Reply:
Ok.